rfcs/field-selection.md

```diff
@@ -0,0 +1,1438 @@
+This RFC proposes a syntax for `FieldSelection` that allows for flexible field selections and reshaping of input objects.
```

Comment:
guess this all got moved to the appendix so should this file be removed to avoid the confusion?